### ISSUE TITLE
[Bug] Monaco 0.49.0 still ships with TS 5.0.2

### ISSUE THREAD

**Issue by spahnke** [microsoft/monaco-editor#4537]
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [ ] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Monaco Editor Playground Link

https://microsoft.github.io/monaco-editor/playground.html?source=v0.49.0#XQAAAAIcAwAAAAAAAABBqQkHQ5NjdMjwa-jY7SIQ9S7DNlzs5W-mwj0fe1ZCDRFc9ws9XQE0SJE1jc2VKxhaLFIw9vEWSxW3yscw26dnCAEiAKctgAboa7AaZZDHLgrLqPRv_fRv2RyIokydlTdTYccsu6v6jDrc8jQ8ulrCVzjWBVadhzlM4CndfaRo4TdtOngMwKEugNUWDywuRgUpL9GqimHxLC4ufkji7j8557XjTAHG271EXRi9DODuwGeE5hhOACSVwod70zJSA8sYe4BAeVI3cqK-4pwSWk2Cp1aJF5Vjh3cP4pjhEu6dbemGiVgyvSUPw1o5VHXVKZT-gUzuyQQdv6Ycy755BK1oIhTom2WWhRDVN6CY1FbDBxiNjlSLrZJaa2Rfpia1RllBuwM4A5kV2pPz-0tD9hMePfslJQe21CLQu70hi2Obb-doOE5ifO8P_Gffj0Ucczr8VJ2PbxCM-5v7SsirAsxqx8Hg8KhrGherxxXTrOcLi4wCrbHU4X-ue-5XmkbGKJBcCaA689RdOem9KLNxn7YU1rPnV3JGLGefhNhR81EjNkaIVjCjPoWBl0ekxXsWrkh1Jx5FEYLfWn6UK-uP1_DcxjWAhiRcd5v7anfYZz9li6ceCo9VAnby13mBsejx9wiQ4LkP8nJE4TR-v27IS_W_9r__-s5YCA

### Monaco Editor Playground Code

```typescript
const value = /* set from `myEditor.getModel()`: */ `function getUrls(url: string | URL, names: string[]) {
    if (typeof url === "string") {
        url = new URL(url);
    }

    return names.map(name => {
        url.searchParams.set("name", name)
        //  ~~~~~~~~~~~~
        // error!
        // Property 'searchParams' does not exist on type 'string | URL'.
        // this is fixed in TS 5.4 and shouldn't show an error anymore

        return url.toString();
    });
}
`;

// Hover on each property to see its docs!
const myEditor = monaco.editor.create(document.getElementById("container"), {
	value,
	language: "typescript",
	automaticLayout: true,
});

console.log(monaco.languages.typescript.typescriptVersion);
```

### Actual (Problematic) Behavior

- `searchParams` is still marked as an error although TS 5.4 fixed that behavior (https://devblogs.microsoft.com/typescript/announcing-typescript-5-4/#preserved-narrowing-in-closures-following-last-assignments)
- the version logged is 5.0.2

### Expected Behavior

- no error marked in the code
- the version logged is 5.4.x

### Additional Context

After #4305 was merged I was under the impression that the next version of Monaco (i.e. 0.49.x) will include the new TS version automatically. Is that not the case?

**Comment by jakebailey** [comment]
I'm pretty sure it ships with 5.4, it's just that the default options don't raise the lib option to one that includes that prop: https://github.com/microsoft/monaco-editor/blob/a5e379c985d3263aacb1e811958c4663fea3db67/src/language/typescript/monaco.contribution.ts#L753

The TS playground overrides these options and so doesn't have this problem.

**Comment by spahnke** [comment]
This is a) a compiler feature though, not a feature of an updated library and b) the console log reports 5.0.2 as the shipped version. The same playground code above works with both `v0.49.0-dev-20240524` and `v0.50.0-dev-20240530` where the error is gone and the console log reports 5.4.5. But it breaks/reverts back to 5.0.2 with both `0.49.0-rc` and `0.49.0`.

**Comment by jakebailey** [comment]
Oh, I didn't notice the console log 😞

**Comment by jakebailey** [comment]
@hediet @aiday-mar Did something go wrong in publishing? Looking at https://unpkg.com/browse/monaco-editor@0.49.0/esm/vs/language/typescript/ts.worker.js, the published package definitely contains TS 5.0.

```
  var init_corePublic = __esm({
    "src/compiler/corePublic.ts"() {
      "use strict";
      versionMajorMinor = "5.0";
      version = "5.0.2";
      Comparison = /* @__PURE__ */ ((Comparison3) => {
        Comparison3[Comparison3["LessThan"] = -1] = "LessThan";
        Comparison3[Comparison3["EqualTo"] = 0] = "EqualTo";
        Comparison3[Comparison3["GreaterThan"] = 1] = "GreaterThan";
        return Comparison3;
      })(Comparison || {});
    }
  });
```

**Comment by spahnke** [comment]
TS 5.4.5 ships now with 0.50.0
